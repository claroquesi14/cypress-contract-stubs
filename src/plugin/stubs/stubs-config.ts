export interface LocalStub {
  file: string;
  mode?: 'local';
  path?: string;
}

export interface RemoteStub {
  id: string;
  mode?: 'remote';
  type?: 'nexus' | 'nexus3';
  server?: string;
  repository?: string;
}

export let configVars = {
  mavenRepository: '.m2/repository',
  cachePath: 'node_modules/.cache/stubs',
  server: 'http://localhost:8081',
  repository: 'maven-releases',
  endpointNexus3: 'service/rest/v1/search/assets',
  endpointNexus3Context: '',
  endpointNexus: 'service/local/artifact/maven/resolve',
  endpointNexusRepositories: 'service/local/repositories',
  endpointNexusContext: ''
};

export let localStubs: LocalStub[] = [];
export let remoteStubs: RemoteStub[] = [];

/**
 * Parse plugin configuration
 *
 * @param env
 */
// eslint-disable-next-line
export function parseConfiguration(env: Record<string, any>): void {
  const stubs = Array.isArray(env.stubs) ? env.stubs : [];
  localStubs = stubs.filter((stub) => stub.mode === 'local');
  remoteStubs = stubs.filter((stub) => stub.mode === undefined || stub.mode === 'remote');

  configVars = {
    mavenRepository: env.stubs_maven_repo || configVars.mavenRepository,
    cachePath: env.stubs_cache_path || configVars.cachePath,
    server: env.stubs_server || configVars.server,
    repository: env.stubs_repository || configVars.repository,
    endpointNexus3: env.stubs_endpoint_nexus3 || configVars.endpointNexus3,
    endpointNexus3Context: env.stubs_endpoint_nexus3_context || configVars.endpointNexus3Context,
    endpointNexus: env.stubs_endpoint_nexus || configVars.endpointNexus,
    endpointNexusRepositories: env.stubs_endpoint_nexus_repos || configVars.endpointNexusRepositories,
    endpointNexusContext: env.stubs_endpoint_nexus_context || configVars.endpointNexusContext
  };
}

/**
 * Retrieve component coordinates from artifact id
 * @param id
 */
export function stubCoordinate(id: string): {
  groupId: string;
  artifactId: string;
  version: string;
  classifier: string;
} {
  const [groupId, artifactId, version, classifier] = id.split(':');
  return { groupId, artifactId, version, classifier };
}
