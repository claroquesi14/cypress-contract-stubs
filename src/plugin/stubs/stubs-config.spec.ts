import { configVars, localStubs, parseConfiguration, remoteStubs } from './stubs-config';
import { expect } from '@jest/globals';

describe('StubConfig Test', () => {
  const stubs = [{ mode: 'remote' }, { mode: 'local' }, { mode: 'remote' }];

  const customVars = {
    stubs_maven_repo: 'maven/.m2/repository',
    stubs_cache_path: 'dist/stubs',
    stubs_server: 'http://127.0.0.1:3333',
    stubs_repository: 'main-releases',
    stubs_endpoint_nexus3: 'service/rest/v2/search/assets',
    stubs_endpoint_nexus3_context: 'nexus3',
    stubs_endpoint_nexus: 'service/proxy/artifact/maven/resolve',
    stubs_endpoint_nexus_repos: 'proxy/service/local/repositories',
    stubs_endpoint_nexus_context: 'nexus'
  };

  it('should return default configuration variables', () => {
    expect(configVars).toMatchSnapshot();
  });

  it('should parse plugin configuration variables', () => {
    parseConfiguration(customVars);
    expect(configVars).toMatchSnapshot();
  });

  it('should filter and store local and remote stubs', () => {
    parseConfiguration({ stubs });
    expect(localStubs.length).toBe(1);
    expect(remoteStubs.length).toBe(2);
  });
});
