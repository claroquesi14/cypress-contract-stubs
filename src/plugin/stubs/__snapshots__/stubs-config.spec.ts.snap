// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StubConfig Test should parse plugin configuration variables 1`] = `
Object {
  "cachePath": "dist/stubs",
  "endpointNexus": "service/proxy/artifact/maven/resolve",
  "endpointNexus3": "service/rest/v2/search/assets",
  "endpointNexus3Context": "nexus3",
  "endpointNexusContext": "nexus",
  "endpointNexusRepositories": "proxy/service/local/repositories",
  "mavenRepository": "maven/.m2/repository",
  "repository": "main-releases",
  "server": "http://127.0.0.1:3333",
}
`;

exports[`StubConfig Test should return default configuration variables 1`] = `
Object {
  "cachePath": "node_modules/.cache/stubs",
  "endpointNexus": "service/local/artifact/maven/resolve",
  "endpointNexus3": "service/rest/v1/search/assets",
  "endpointNexus3Context": "",
  "endpointNexusContext": "",
  "endpointNexusRepositories": "service/local/repositories",
  "mavenRepository": ".m2/repository",
  "repository": "maven-releases",
  "server": "http://localhost:8081",
}
`;
